doc/holochain_101/src/glossary.md

```diff
@@ -1 +1,142 @@
 # Glossary
+
+<dl>
+   <dt>Absolutism</dt>
+   <dd><p>A view which says there is ONE absolute frame of reference or view of reality. For example, global ledger consensus systems assert the state of the ledger is the one authoritative absolute truth of the system.</p></dd>
+   <dt>Agent</dt>
+   <dd><p>An entity which operates with agency -- meaning a human or a program, which makes
+                    choices outside of whatever code is written into holochain.</p></dd>
+   <dt>Application Engine</dt>
+   <dd><p>(a.k.a. Application Space) (See Nucleus & Ribosome)</p></dd>
+   <dt>Author</dt>
+   <dd><p>The agent who originally created the information on their source chain / author chain.</p></dd>
+   <dt>Base</dt>
+   <dd><p>Provide a hash for the primary resource that you want to define a relation to
+Tag: Provide a string that defines the relationship, and which defines how it should be looked up in the future
+Link: Provide a hash for the secondary resource that you want to associate with the Base
+ It has the side effect of adding the Link+Tag to the base hash on the DHT for each of the links in the list. Thus when you do a getLinks(base,tag) it will return all the links of that tag on that base. This is how we do relations in a sparse hash space, allowing us to convert a hash data store into a graph. Related: the 'commit' method of the holochain API.</p></dd>
+   <dt>Blacklist</dt>
+   <dd><p>A list of nodes who get blocked for attempting to propagate invalid information.</p></dd>
+   <dt>Blockchain</dt>
+   <dd><p>An architecture used by Bitcoin, Ethereum and many cryptocurrencies to decentralize maintenance of a global ledger for token accounting.</p></dd>
+   <dt>CALM</dt>
+   <dd><p>Consistency as Logical Monotonicity. (see Monotonicity)</p></dd>
+   <dt>CAP Theorem</dt>
+   <dd><p>Consistency-Availability-PartitionTolerance. Often depicted as a triangle, where you can only achieve two of three options. Unfortunately in the real-world, you always have to include 'P' because networks are not 100% reliable. So the question often comes down to are you optimzing for data Consistency or data Availablily (making sure new data is not visible to anyone until its visible to everyone). Holochains are eventually consistent, but our notion of Consistency is not that there is one set of absolutely true data, but rather you know exactly what was asserted by whom and when.</p></dd>
```

Comment:
also see discussion at https://en.wikipedia.org/wiki/CAP_theorem and https://en.wikipedia.org/wiki/PACELC_theorem

P is described as an `if` rather than a constant